webapp/src/app/projects/[projectName]/[languageName]/@explorer/default.tsx

```diff
@@ -0,0 +1,2 @@
+import Explorer from './page';
+export default Explorer;
```

Comment:
Maybe!

It's not strictly true that the sidebar will always be rendered identically regardless of the URL. In this pull request it is, but that's a temporary thing due to the implementation being incomplete. There are a couple of important UX enhancements coming.

1.  We're going to pass different values to [`<RichTreView defaultExpandedItems={…} />`](https://mui.com/x/api/tree-view/rich-tree-view#rich-tree-view-prop-defaultExpandedItems) depending on the URL. If you arrive at e.g. `/projects/zetkin/en/feat.email.heading` the tree will already have `feat` and `email` expanded, so that the editing context is just as clear as it would have been if you had clicked through to it yourself.
2. When those expanded nodes are outside the immediately visible area at the top of the sidebar, we'll set the initial scroll position so that they're visible.

The above is why I opted for the more advanced parallel routing setup. I'm going to investigate this by simply adding them to the PR. I'll implement them with the advanced routing setup and then try to scale back to the simpler layout-based setup and see if it's possible to do so without sacrificing on the UX. Will be bloody great if it works out!

Reply:
That’s a good point but are we going to implement that with different page.tsx ?

Reply:
Go check out https://github.com/zetkin/lyra/pull/113! It's a pull request proposing some changes to this pull request, including getting rid of the parallel routing stuff.